automated_analysis.py

```diff
@@ -367,7 +367,44 @@ def set_survey_percentages(survey_counts, total_survey_counts):
 
         for sample in samples:
             writer.writerow(sample)
-    
+
+    # Export raw messages labelled with Meta gratitude and about conversation programmatically known as success_story
+    log.info("Exporting success story raw messages for each episode...")
+    success_story_messages = []  # of dict of code_string_value to avf-uid and raw messages
+    success_story_string_values = ["gratitude","about_conversation"]
+    for plan in PipelineConfiguration.RQA_CODING_PLANS:
+        for cc in plan.coding_configurations:
+            success_story_code_to_messages = OrderedDict()
+            for string_value in success_story_string_values:
+                success_story_code_to_messages[string_value] = []
+
+            for msg in messages:
+                if not AnalysisUtils.labelled(msg, CONSENT_WITHDRAWN_KEY, plan):
+                    continue
+
+                for label in msg[cc.coded_field]:
+                    code = cc.code_scheme.get_code_with_code_id(label["CodeID"])
+
+                    if code.string_value in success_story_string_values:
+                        success_story_code_to_messages[code.string_value].append([msg['uid'], msg[plan.raw_field]])
+
+            for code_string_value in success_story_string_values:
```

Comment:
Do you need to do this in 2 phases or can you just write directly to success_story_messages without needing to use success_story_code_to_messages first?

Reply:
refactored to loop once without the success_story_code_to_messages step. Thanks